### PR TITLE
Better GPU Documentation

### PR DESCRIPTION
**Pull request by LLukas22**
Moved GPU Documentation to `acceleration-support.md` and added documentation on how to enable gpu support and which model architectures are supported. 

[Rendered Version](https://github.com/LLukas22/llm/blob/chore/gpu-documentation/doc/acceleration-support.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ opt-level = 3
 
 The `llm` library is engineered to take advantage of hardware accelerators such as `cuda` and `metal` for optimized performance.
 
-To enable `llm` to harness these accelerators, some preliminary configuration steps are necessary, which vary based on your operating system. For comprehensive guidance, please refer to the [Acceleration Support for Building section](doc/CONTRIBUTING.md#acceleration-support-for-building) in our documentation.
+To enable `llm` to harness these accelerators, some preliminary configuration steps are necessary, which vary based on your operating system. For comprehensive guidance, please refer to the [Acceleration Support for Building section](doc/acceleration-support.md) in our documentation.
```
Comment (philpax) [inline README.md:103]: Remove `for Building section` in the linktext

File: doc/acceleration-support.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
+# Acceleration Support
+
+The `ggml-sys` crate includes various acceleration backends, selectable via `--features` flags. The availability of supported backends varies by platform, and `ggml-sys` can only be built with a single active acceleration backend at a time. If cublas and clblast are both specified, cublas is prioritized and clblast is ignored.
```
Comment (philpax) [inline doc/acceleration-support.md:3]: Also applies to `llm` etc - maybe say "The `llm` ecosystem of crates, including `llm`, `llm-base` and `ggml` support various acceleration backends"

Comment (philpax) [reply doc/acceleration-support.md:3]: CuBLAS and CLBlast (consistent capitalisation is important when referring to the solutions and not the features)

File: doc/acceleration-support.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+| Platform/OS | `cublas`           | `clblast`          | `metal`            |
+| ----------- | ------------------ | ------------------ | ------------------ |
+| Windows     | :heavy_check_mark: | :heavy_check_mark: | :x:                |
+| Linux       | :heavy_check_mark: | :heavy_check_mark: | :x:                |
+| MacOS       | :x:                | :x:                | :heavy_check_mark: |
+
+## Utilizing GPU Support
+
+To activate GPU support, set the `use_gpu` attribute of the `ModelParameters` to `true`.
```
Comment (philpax) [inline doc/acceleration-support.md:13]: "To activate GPU support (assuming that you have enabled one of the features above),"

File: doc/acceleration-support.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+To activate GPU support, set the `use_gpu` attribute of the `ModelParameters` to `true`. 
+
+- **CLI Users:** You can enable GPU support by adding the `--use-gpu` flag.
+  
+- **Backend Consideration:** For users leveraging the `cublas` or `clblast` backends, you can specify the number of layers you wish to offload to your GPU with the `gpu_layers` parameter in the `ModelParameters`. By default, all layers are offloaded. However, if your model size exceeds your GPU's VRAM, you can specify a limit, like `20`, to offload only the first 20 layers. For CLI users, this can be achieved using the `--gpu-layers` parameter.
+
+**Example:** To run a `llama` model with CUDA acceleration and offload all its layers, your CLI command might resemble:
+
+```bash
+cargo run --release --features cublas -- infer -a llama -m [path/to/model.bin] --use-gpu -p "Help a Lama is standing in my garden!"
```
Comment (philpax) [inline doc/acceleration-support.md:22]: llama, not lama

File: doc/acceleration-support.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+```
+
+💡 **Pro Tip:** For those with ample VRAM using `cublas` or `clblast`, you can significantly reduce your prompt's feed time by increasing the batch size, for example, to `256` or `512` (default is `8`).
```
Comment (philpax) [inline doc/acceleration-support.md:25]: `batch size; for example` instead of `batch size, for example`

**Review by philpax — CHANGES_REQUESTED** [review]
LGTM, just some minor grammar nits but otherwise merge once fixed

**Comment by LLukas22** [comment]
@philpax could you apply your proposed changes and have @danforbes take a final look?

**Pull request merged**